.github/actions/require-multiple-reviewers-v1/src/utils.ts

```diff
@@ -0,0 +1,40 @@
+import fs from 'fs'
+import ignore from 'ignore'
+import { Annotation } from './types'
+
+/**
+ * Returns an array of annotations for the important files that require multiple reviewers.
+ * @param importantFilesChanged
+ * @returns Array<Annotation>
+ */
+export function getAnnotations(
+  importantFilesChanged: Array<string>,
+  reviewersNumber: number
+): Array<Annotation> {
+  return importantFilesChanged.map(file => ({
+    path: file,
+    start_line: 1,
+    end_line: 1,
+    annotation_level: 'warning',
+    message: `The file ${file} is important and requires at least ${reviewersNumber} reviewers.`
+  }))
+}
+
+/**
+ * Returns an array of important files that have changed.
+ * @param IMPORTANT_FILES_PATH
+ * @param changedFiles
+ * @returns Array<string>
+ */
+export function getImportantFilesChanged(
+  IMPORTANT_FILES_PATH: string,
+  changedFiles: string[]
+): string[] {
+  // Since this is the ignore package, it will filter out files that match the patterns
+  const notImportantFiles = ignore()
+    .add(fs.readFileSync(IMPORTANT_FILES_PATH, 'utf-8').toString())
+    .filter(changedFiles)
+
+  // Get all the files that were filtered out earlier - those are the important files
+  return changedFiles.filter(file => !notImportantFiles.includes(file))
```

Comment:
This is `O(unimportant_file_count * total_changed_file_count)`, which might be very large for a PR that touches lots of unimportant files. Can you convert `notImportantFiles` to a `Set` in between L36 and L39 to avoid the quadratic worst-case?

Reply:
Good point, done some reading about it too :D I updated the logic to do less filtering, let me know if any array still should be a `Set` but I am not sure how it would help in this case.